scripts/ci/lib.sh

```diff
@@ -63,6 +63,22 @@ ci_exit_trap() {
         set_ci_shared_export JOB_DISPATCH_OUTCOME "${OUTCOME_FAILED}"
     fi
 
+    if [[ -f "${SHARED_DIR:-}/shared_env" ]]; then
+        # shellcheck disable=SC1091
+        cat "${SHARED_DIR:-}"/shared_env
+        source "${SHARED_DIR:-}/shared_env"
+    fi
+
+    save_job_record "${JOB_NAME:-missing}" "prow" \
```

Comment:
Would it make sense to to define this as a function and use both here and in `.openshift-ci/end.sh`?